src/main/java/org/arl/fjage/param/NamedParameter.java

```diff
@@ -74,20 +74,22 @@ public String toString() {
   }
 
   /**
-   * Compares if two named parameters are equals.
+   * Compares if a parameter is equivalent to the named parameter.
    *
    * @return true if two parameters are equal
    */
   @Override
   public boolean equals(Object obj) {
-    if (!(obj instanceof NamedParameter)) return false;
-    NamedParameter p = (NamedParameter)obj;
+    if (!(obj instanceof Parameter)) return false;
     String name1 = name;
     if (name1.contains(".")) name1 = name1.substring(name1.indexOf('.')+1);
```

Comment:
This only peels of the first item of the path. 

```groovy
groovy:000> name = "org.arl.unet.nodeinfo.NodeInfoParams.address"

groovy:000> name.substring(name.indexOf('.')+1)
===> arl.unet.nodeinfo.NodeInfoParams.address
```

Isn't this what we want here?

```groovy
groovy:000> name.substring(name.lastIndexOf('.')+1)
===> address
```

Reply:
We don't use qualified named parameters, so this doesn't make sense. Since I don't recall the exact context in which this code was added (see comment below), I am reluctant to change this in this PR. If it's genuinely a bug, it should be fixed in its own PR.

Reply:
Ah, I guess I misunderstood you when we talked about this on the UnetStack call last week. I thought `NamedParameter`s should support fully qualified paths as well. 

In this case, my only concern with this PR is that we are abusing `equals()` in ways it wasn't intended for. Arguably a bit pedantic, but I'd say better be safe than sorry?

Reply:
My initial reaction was that the abuse of `equal()` makes the design simpler and so might be OK. But on deeper thought, you're correct in pointing out that there may be issues with Sets and Maps, as there are no guarantees of which order the comparisons will be made. I have reverted back the `equals()` to only handle named parameter comparisons and return `false` for other comparisons.